proposals/lip-0052.md

```diff
@@ -679,6 +679,168 @@ else: # chainID == sendingChainID
+   * For all elements of the `NFTSubstore` array, if the value of `owner` has length 4 (representing an escrowed NFT) then the corresponding `NFTID.chainID` must equal `CHAIN_ID_ALIAS_NATIVE`.
+   * For all elements of the `NFTSubstore` array, if the value of `attributes` has length less than or equal to `MAX_BYTE_LENGTH_ATTRIBUTES`.
+   * Accross all element of the `collectionSubstore` array, all values given for `collection` must be unique.
+
+* For each entry `NFTEntry` in `genesisBlockAssetObject.NFTSubstore`, create an entry in the NFT substore  with
+  ```python
+  storeKey = uint32be(NFTEntry.NFTID.chainID) || uint32be(NFTEntry.NFTID.collection) || uint64be(NFTEntry.NFTID.index)
+  storeValue = {"owner": NFTEntry.owner,
+                "lockingModuleID": NFTEntry.lockingModuleID,
+                "attributes": NFTEntry.attributes
+               } serialized using NFTStoreSchema.
+  ```
+  Further, if `NFTEntry.owner` has length 20, create an entry in the user substore with
+  ```python
+  storeKey = NFTEntry.owner || uint32be(NFTEntry.NFTID.chainID) || uint32be(NFTEntry.NFTID.collection) || uint64be(NFTEntry.NFTID.index)
+  storeValue = {"exists": True} serialized using userStoreSchema.
+  ```
+
+* For each entry `collectionEntry` in `genesisBlockAssetObject.collectionSubstore`, create an entry in the collection substore  with
+  ```python
+  storeKey = uint32be(collectionEntry.collection)
+  storeValue = {"nextAvailableIndex": collectionEntry.nextAvailableIndex} serialized using collectionStoreSchema.
+  ```
+
+* Create an entry in the available collection substore with
+  ```python
+  storeKey = EMPTY_BYTES
+  storeValue = {"nextAvailableCollection": genesisBlockAssetObject.availableCollectionSubstore} 
+               serialized using availableCollectionStoreSchema.
+  ```
+
+* For each entry `terminatedChainID` in `genesisBlockAssetObject.terminatedEscrowSubstore`, create an entry in the terminated escrow substore with
+  ```python
+  storeKey = uint32be(terminatedChainID)
+  storeValue = {"escrowTerminated": True} serialized using terminatedEscrowStoreSchema.
+  ```
+  
+Once the module store is initialized, its validity is attested asserting that the function below returns `True`.
+```python
+NFTStoreValid():
+    existingIndex = {}
+    for storeKey a store key in NFT substore:
+        chainID = storeKey[0:4] # part of the key corresponding to the chain ID of the NFT
+        collection = storeKey[4:8] # part of the key corresponding to the collection of the NFT
+        index = storeKey[8:12] # part of the key corresponding to the index of the NFT
+        if chainID == CHAIN_ID_ALIAS_NATIVE:
+            existingIndex[collection] = max{existingIndex[collection], index}
+    
+    # check that all existing indices are smaller than the next available index for the collection
+    for collection in existingIndex:
+        nextAvailableIndex = store value of the collection substore entry with store key 'collection'
```

Comment:
just to clarify, we are assigning the value stored in the property `nextAvailableIndex` of the collection substore to a property called `nextAvailableIndex` here?

Reply:
yes, it's a bit weird. Changed to use the already defined notation.